src/uu/du/src/du.rs

```diff
@@ -318,6 +320,16 @@ fn du(
                 Ok(entry) => match Stat::new(entry.path()) {
                     Ok(this_stat) => {
                         if this_stat.is_dir {
+                            if options.one_file_system
+                                && this_stat.inode.is_some()
+                                && my_stat.inode.is_some()
```

Comment:
You can rewrite this nicely with `if let`, so that you don't need `unwrap` and it's clear that they won't panic:
```rust
if options.one_file_system {
    if let (Some(this_inode), Some(my_inode)) = (this_stat.inode, my_stat.inode) {
       ...
    }
}

Reply:
Updated